src/i965_gpe_utils.c

```diff
@@ -1406,6 +1406,27 @@ i965_object_surface_to_2d_gpe_resource(struct i965_gpe_resource *res,
 }
 
 void
+i965_object_surface_to_2d_gpe_resource_with_align(struct i965_gpe_resource *res,
+                                                  struct object_surface *obj_surface)
+{
+    unsigned int swizzle;
```

Comment:
Directly change the surface value(orig_width ,orig_height) looks not good. Here GPE resource is the birdge to attend the final operation. it is better to do alignment with gpe resource. The issue maybe be there when the post process on this surface.
BTW, the best align operation should be done when obj surface is allocated . But this is implemented in the middle ware. For driver, it only work around.